### ISSUE TITLE
Update Data Structure

### ISSUE THREAD

**Issue by simlu** [simlu/voxelshop#72]
This is quite a big task. Currently every voxels is reflected by an object. This only works for small creations.

This combined with using a graphic rendering engine would mean we can support large objects.

**Comment by simlu** [comment]
I've already started designing this. Have to go over req. list more.
Useful and important: organize world in e.g. 32x32 chunks

This is an example idea (probably not very good yet!):
- organize world in e.g. 32x32 chunks
- only chunks that are filled are created
-> empty chunks are deleted
- store voxel types in world array HashMap<short, Voxel> that maps voxel type to voxel type details
- each chunk is compressed by RLE, e.g. voxelID + count and so on (might not even be needed)
- the RLE is indexed with a binary tree to efficiently access a certain memory location

**Comment by MrTopom** [comment]
Maybe keep in mind the fact that some plugin could save things in the file also. So make a structure able to store some optional plug-in related data.

**Comment by simlu** [comment]
Yeah the new file structure will be very flexible to allow for these things. Will have a google document where we can talk design!
